src/cmd/cli/command/commands_test.go

```diff
@@ -140,84 +139,99 @@ func TestVersion(t *testing.T) {
 func TestCommandGates(t *testing.T) {
 	mockService := &mockFabricService{}
 	_, handler := defangv1connect.NewFabricControllerHandler(mockService)
+	origDir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Failed to get current directory: %v", err)
+	}
+	composeDir := "../../../../src/testdata/sanity"
+	os.Chdir(composeDir)
 
 	server := httptest.NewServer(handler)
-	t.Cleanup(server.Close)
-
-	t.Setenv("AWS_REGION", "us-test-2")
+	t.Cleanup(func() {
+		os.Chdir(origDir)
+		server.Close()
+	})
 
-	if err := os.Chdir("../../../testdata/empty"); err != nil {
-		t.Fatalf("Failed to change directory: %v", err)
+	type cmdPermTest struct {
+		name      string
+		command   []string
+		errorText string
+		wantError error
 	}
+	type cmdPermTests []cmdPermTest
 
-	testData := []struct {
-		name          string
-		command       []string
-		accessAllowed bool
-		wantError     string
-	}{
+	t.Setenv("AWS_REGION", "us-test-2")
+
+	testData := cmdPermTests{
 		{
-			name:          "compose up - aws - no access",
-			command:       []string{"compose", "up", "--provider=aws", "--dry-run"},
-			accessAllowed: false,
-			wantError:     "current subscription tier does not allow this action: no access to use aws provider",
+			name:      "compose up - aws - no access",
+			command:   []string{"compose", "up", "--project-name=app", "--provider=aws", "--dry-run"},
+			errorText: "no access to use aws provider",
+			wantError: connect.NewError(connect.CodeResourceExhausted, errors.New("no access to use aws provider")),
```

Comment:
This feels weird, like you're testing your test mock code, ie. you return `errorText` and then test that you got that error from the mock.